downloads-generation/cross_validation_class1/split_folds.py

```diff
@@ -0,0 +1,113 @@
+"""
+Split training data into CV folds.
+"""
+import argparse
+import sys
+from os.path import abspath
+
+import pandas
+from sklearn.model_selection import StratifiedKFold
+
+parser = argparse.ArgumentParser(usage = __doc__)
+
+parser.add_argument(
+    "input", metavar="INPUT.csv", help="Input CSV")
+
+parser.add_argument(
+    "--folds", metavar="N", type=int, default=5)
+
+parser.add_argument(
+    "--allele",
+    nargs="+",
+    help="Include only the specified allele(s)")
+
+parser.add_argument(
+    "--min-measurements-per-allele",
+    type=int,
+    metavar="N",
+    help="Use only alleles with >=N measurements.")
+
+parser.add_argument(
+    "--subsample",
+    type=int,
+    metavar="N",
+    help="Subsample to first N rows")
+
+parser.add_argument(
+    "--random-state",
+    metavar="N",
+    type=int,
+    help="Specify an int for deterministic splitting")
+
+parser.add_argument(
+    "--output-pattern-train",
+    default="./train.fold_{}.csv",
+    help="Pattern to use to generate output filename. Default: %(default)s")
+
+parser.add_argument(
+    "--output-pattern-test",
+    default="./test.fold_{}.csv",
+    help="Pattern to use to generate output filename. Default: %(default)s")
+
+
+def run(argv):
+    args = parser.parse_args(argv)
+
+    df = pandas.read_csv(args.input)
+    print("Loaded data with shape: %s" % str(df.shape))
+
+    df = df.ix[
```

Comment:
What's the difference between `df.ix[]` and `df[]` in this case?

Reply:
`df[]` is for selecting columns, `df.ix` (or `df.loc`, which is now recommended) is for selecting rows (and optionally also columns). As a backward compatibility thing you can use `df[]` with a binary mask to select rows but that is deprecated. I updated this to be `df.loc` instead of `df.ix`.